src/pretix/base/models/orders.py

```diff
@@ -229,7 +229,14 @@ def can_modify_answers(self) -> bool:
 
     @property
     def can_user_cancel(self) -> bool:
-        return self.event.settings.cancel_allow_user
+        positions = self.positions.all().prefetch_related('item')
```

Comment:
I'd prefer ``select_related`` over ``prefetch_related`` in this case as it will cause one query less. Would you mind changing this?